packages/components/pie-modal/src/modal.scss

```diff
@@ -181,16 +177,15 @@
     }
 
     // Ensure correct padding when there is a back button in front of the heading
-    &[hasbackbutton] .c-modal-heading {
+    .c-modal-backBtn + .c-modal-heading {
```

Comment:
The back button already has a class, so we can use that and the native next-sibling selector instead of having a whole other class to achieve the same thing